File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/tapaislands/android/TapaIslandsMainActivity.java
package com.zwstudio.logicpuzzlesandroid.puzzles.tapaislands.android;

import com.zwstudio.logicpuzzlesandroid.R;
import com.zwstudio.logicpuzzlesandroid.common.android.GameMainActivity;
import com.zwstudio.logicpuzzlesandroid.puzzles.tapaislands.data.TapaIslandsDocument;
import com.zwstudio.logicpuzzlesandroid.puzzles.tapaislands.domain.TapaIslandsGame;
import com.zwstudio.logicpuzzlesandroid.puzzles.tapaislands.domain.TapaIslandsGameMove;
import com.zwstudio.logicpuzzlesandroid.puzzles.tapaislands.domain.TapaIslandsGameState;

import org.androidannotations.annotations.Bean;
import org.androidannotations.annotations.Click;
import org.androidannotations.annotations.EActivity;

@EActivity(R.layout.activity_game_main)
public class TapaIslandsMainActivity extends GameMainActivity<TapaIslandsGame, TapaIslandsDocument, TapaIslandsGameMove, TapaIslandsGameState> {
    @Bean
    protected TapaIslandsDocument document;
    public TapaIslandsDocument doc() {return document;}

    @Click
    void btnOptions() {
        TapaIslandsOptionsActivity_.intent(this).start();
    }

    protected void resumeGame() {
        doc().resumeGame();
        TapaIslandsGameActivity_.intent(this).start();
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/tapaislands/android/TapaIslandsMainActivity.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.tapaislands.androidimport

import com.zwstudio.logicpuzzlesandroid.R
import com.zwstudio.logicpuzzlesandroid.common.android.GameMainActivity
import com.zwstudio.logicpuzzlesandroid.puzzles.tapaislands.data.TapaIslandsDocument
import com.zwstudio.logicpuzzlesandroid.puzzles.tapaislands.domain.TapaIslandsGame
import org.androidannotations.annotations.Bean
import org.androidannotations.annotations.EActivity

_
@EActivity(R.layout.activity_game_main)
class TapaIslandsMainActivity : GameMainActivity<TapaIslandsGame?, TapaIslandsDocument?, TapaIslandsGameMove?, TapaIslandsGameState?>() {
    @Bean
    protected var document: TapaIslandsDocument? = null
    override fun doc() = document

    @Click
    fun btnOptions() {
        TapaIslandsOptionsActivity_.intent(this).start()
    }

    protected override fun resumeGame() {
        doc().resumeGame()
        TapaIslandsGameActivity_.intent(this).start()
    }
}

@EActivity(R.layout.activity_game_options)
class TapaIslandsOptionsActivity : GameOptionsActivity<TapaIslandsGame?, TapaIslandsDocument?, TapaIslandsGameMove?, TapaIslandsGameState?>() {
    @Bean
    protected var document: TapaIslandsDocument? = null
    override fun doc() = document
}

@EActivity(R.layout.activity_game_help)
class TapaIslandsHelpActivity : GameHelpActivity<TapaIslandsGame?, TapaIslandsDocument?, TapaIslandsGameMove?, TapaIslandsGameState?>() {
    @Bean
    protected var document: TapaIslandsDocument? = null
    override fun doc() = document
}